### ISSUE TITLE
[Bug]: panic query on select count(*) from ..

### ISSUE THREAD

**Issue by millken** [hydradatabase/hydra#168]
### What's wrong?

I used pg_ivm to synchronize a table to columnar. When testing, I found that the cloumnar table count is very slow, and even panic can be seen in the log (maybe triggered when Ctrl +C)
```
mainnet=# \d+ reader.staking_actions;
                                                Table "reader.staking_actions"
    Column     |         Type          | Collation | Nullable | Default | Storage  | Compression | Stats target | Description 
---------------+-----------------------+-----------+----------+---------+----------+-------------+--------------+-------------
 id            | bigint                |           |          |         | plain    |             |              | 
 block_height  | bigint                |           |          |         | plain    |             |              | 
 bucket_id     | bigint                |           |          |         | plain    |             |              | 
 owner_address | character varying(42) |           |          |         | extended |             |              | 
 candidate     | character varying(42) |           |          |         | extended |             |              | 
 amount        | numeric(42,0)         |           |          |         | main     |             |              | 
 act_type      | character varying(42) |           |          |         | extended |             |              | 
 sender        | character varying(42) |           |          |         | extended |             |              | 
 act_hash      | text                  |           |          |         | extended |             |              | 
 auto_stake    | boolean               |           |          |         | plain    |             |              | 
 duration      | bigint                |           |          |         | plain    |             |              | 
Indexes:
    "reader_idx_staking_actions_block_height" btree (block_height)
    "reader_idx_staking_actions_bucket_id" btree (bucket_id)
    "staking_actions_index" UNIQUE, btree (id)
Access method: columnar

mainnet=# \timing
Timing is on.
mainnet=# select count(*) from reader.staking_actions;
  count   
----------
 15461576
(1 row)

Time: 114561.246 ms (01:54.561)
```

log:
```
2023-10-07 01:33:44.962 UTC [2301598] postgres@mainnet ERROR:  canceling statement due to user request
2023-10-07 01:33:44.962 UTC [2301598] postgres@mainnet STATEMENT:  select count(*) from reader.staking_actions;
2023-10-07 01:35:25.549 UTC [2302149] PANIC:  stuck spinlock detected at AdvanceStripeRead, columnar_reader.c:810
2023-10-07 01:35:25.549 UTC [2302149] STATEMENT:  select count(*) from reader.staking_actions;
2023-10-07 01:35:35.463 UTC [2302150] PANIC:  stuck spinlock detected at AdvanceStripeRead, columnar_reader.c:810
2023-10-07 01:35:35.463 UTC [2302150] STATEMENT:  select count(*) from reader.staking_actions;
2023-10-07 01:35:41.751 UTC [2302147] PANIC:  stuck spinlock detected at AdvanceStripeRead, columnar_reader.c:810
2023-10-07 01:35:41.751 UTC [2302147] STATEMENT:  select count(*) from reader.staking_actions;
2023-10-07 01:35:51.202 UTC [2302146] PANIC:  stuck spinlock detected at AdvanceStripeRead, columnar_reader.c:810
2023-10-07 01:35:51.202 UTC [2302146] STATEMENT:  select count(*) from reader.staking_actions;
2023-10-07 01:35:57.400 UTC [2302148] PANIC:  stuck spinlock detected at AdvanceStripeRead, columnar_reader.c:810
2023-10-07 01:35:57.400 UTC [2302148] STATEMENT:  select count(*) from reader.staking_actions;
2023-10-07 01:35:58.147 UTC [2302152] PANIC:  stuck spinlock detected at AdvanceStripeRead, columnar_reader.c:810
2023-10-07 01:35:58.147 UTC [2302152] STATEMENT:  select count(*) from reader.staking_actions;
2023-10-07 01:36:04.139 UTC [2302151] PANIC:  stuck spinlock detected at AdvanceStripeRead, columnar_reader.c:810
2023-10-07 01:36:04.139 UTC [2302151] STATEMENT:  select count(*) from reader.staking_actions;
2023-10-07 01:36:05.553 UTC [1277] LOG:  background worker "parallel worker" (PID 2302150) was terminated by signal 6: Aborted
2023-10-07 01:36:05.553 UTC [1277] DETAIL:  Failed process was running: select count(*) from reader.staking_actions;
2023-10-07 01:36:05.553 UTC [1277] LOG:  terminating any other active server processes
2023-10-07 01:36:10.618 UTC [1277] LOG:  issuing SIGKILL to recalcitrant children
2023-10-07 01:36:10.672 UTC [1277] LOG:  all server processes terminated; reinitializing
2023-10-07 01:36:10.907 UTC [2302684] LOG:  database system was interrupted; last known up at 2023-10-07 01:23:46 UTC
2023-10-07 01:36:11.048 UTC [2302684] LOG:  database system was not properly shut down; automatic recovery in progress
2023-10-07 01:36:11.487 UTC [2302684] LOG:  invalid record length at 8E/1BB28F40: wanted 24, got 0
2023-10-07 01:36:11.487 UTC [2302684] LOG:  redo is not required
```

Update: I deleted the table and use pg_ivm rebuild, the bug is gone

**Comment by mkaruza** [comment]
Hi @millken thank you for report. It would be valuable to us if you can provide some extra information about state of columnar table. Information that we would like to see is number of "stripes" that are created for columnar table. 

That information can be found in `columnar.stripe` metadata table. Unfortunately, that id is internal number and to obtain it from postgresql you need to execute following statements:

`CREATE FUNCTION columnar_relation_storageid(relid oid) RETURNS bigint
    LANGUAGE C STABLE STRICT
    AS 'columnar', $$columnar_relation_storageid$$;`

Declaring this function you can now found internal id for postgresql table oid.

To get number of stripe records for specific table:

`SELECT count(*) FROM columnar.stripe WHERE storage_id =  columnar_relation_storageid('reader.staking_actions'::regclass::oid);`

Thanks,

**Comment by millken** [comment]
> `SELECT count(*) FROM columnar.stripe WHERE storage_id = columnar_relation_storageid('reader.staking_actions'::regclass::oid);`

68465

> 2. Is autovacuum enabled ?

 off

**Comment by mkaruza** [comment]
@millken thanks, that explains and confirms our suspicions what could be problem.

**Comment by wuputah** [comment]
fixed by #170, please give it a go.

PS. would highly recommend turning autovacuum on, or manually running `vacuum` on columnar tables periodically to combine stripes.
